.github/workflows/markdown-lint-fix.yml

```
@@ -0,0 +1,46 @@
+name: Create Markdownlint auto-fix PR
+on:
+  schedule:
+    - cron: "0 0 * * *"
+  workflow_dispatch:
+
+jobs:
+  fix:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Setup Node.js environment
+        uses: actions/setup-node@v3
+        with:
+          node-version: "16"
+
+      - name: Install all yarn packages
+        run: |
+          yarn --frozen-lockfile
+
+      - name: Lint markdown files
+        run: |
+          yarn lint:md
+
+      - name: Create PR with only fixable issues
+        if: success()
+        uses: peter-evans/create-pull-request@v4
+        with:
+          commit-message: "chore: auto-fix Markdownlint issues"
+          branch: daily-markdownlint-fix
```

```suggestion
          branch: daily-markdownlint-fix
          delete-branch: true
```
They've recommended to delete the branch after PR close/merge.

Their document doesn't mention anything about rebasing the exiting branch that has a merged PR already. Ideally everyday we want a fresh branch from main, i.e. `git checkout -b daily-markdownlint-fix main`

---

Let's delete the branch afterward. And yes, let's start with a clean branch from main. Rebasing & co seems error-prone.

If we close the PR without merging, will it delete the branch?

---

> If we close the PR without merging, will it delete the branch?

I strongly believe this happens only when the action decides to close the pull request itself(because recent commit reverted all the changes and there is no diff between main and the branch) or when auto merge is set and after the auto merge.

The action won't delete the branch after humans merge the PR. I think this needs to be handled in a separate task before this one.
For example:
```
- name: Cleanup
    run: |
      echo "deleting the branch daily-markdownlint-fix"
      git branch -D daily-markdownlint-fix
```
This will give us fresh runs everyday instead of appending commits to the same branch.

---

We should set up automatically deleting HEAD.

---

That is better handled by the GitHub setting https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/configuring-pull-request-merges/managing-the-automatic-deletion-of-branches.
If it is closed without merging, the branch will be overridden the next time the job runs.